packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1652,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
 	return inputData;
 }
 
+function findPropertyInArray(
+	name: string,
+	options: INodePropertyMode[],
+): INodePropertyMode | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<INodePropertyOptions | INodeProperties | INodePropertyCollection>,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<
+		INodePropertyOptions | INodeProperties | INodePropertyCollection | INodePropertyMode
+	>,
+):
+	| INodePropertyOptions
+	| INodeProperties
+	| INodePropertyCollection
+	| INodePropertyMode
+	| undefined {
+	const prop = options.find((i) => i.name === name);
+	return prop;
+}
+
+function findPropertyFromParameterName(
+	parameterName: string,
+	nodeType: INodeType,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection {
+	let property: INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+	const paramParts = parameterName.split('.');
+
+	property = findPropertyInArray(paramParts.shift()!, nodeType.description.properties);
+	if (!property) {
+		throw new Error(`Couldn't not find property "${parameterName}"`);
+	}
+
+	// eslint-disable-next-line no-restricted-syntax
+	for (const p of paramParts) {
+		const param = p.split('[')[0];
+		if ('options' in property && property.options) {
```

Comment:
Is `&& property.options` needed?

Reply:
Yes. I get type errors if it's not present.